sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -192,13 +197,12 @@ case class AdaptiveSparkPlanExec(
             stage.resultOption = Some(res)
           case StageFailure(stage, ex) =>
             errors.append(
-              new SparkException(s"Failed to materialize query stage: ${stage.treeString}." +
-                s" and the cause is ${ex.getMessage}", ex))
+              new SparkException(s"Failed to materialize query stage: ${stage.treeString}.", ex))
```

Comment:
If we remove `and the cause is ${ex.getMessage}`, can we get the failed reason ?

Reply:
Yes, we can. `Throwable` will print the cause if any.

Reply:
Yes, but the exception message does not contain the reason, right? If so, when we enable AQE, some exception check ut may fail. The aim to add `and the cause is ${ex.getMessage}` is also to resolve the failed exception check ut when enbale AQE.

Reply:
This is when we need to change the tests.

Reply:
Note: this is the java standard to set the "cause" exception as the "cause", instead of embedding its error message in the current exception.